File: sedmica 3/front-end/src/app/endpoints/student-endpoints/maticna-knjiga/student-maticna-knjiga-get.endpoint.ts
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Injectable} from "@angular/core";
import {MyBaseEndpoint} from "../../MyBaseEndpoint";
import {MojConfig} from "../../../moj-config";
@Injectable({providedIn: 'root'})
export class StudentMaticnaKnjigaGetEndpoint implements  MyBaseEndpoint<number, StudentMaticnaKnjigaGetResponse>{
  constructor(public httpClient:HttpClient) { }

  obradi(studentid: number): Observable<StudentMaticnaKnjigaGetResponse> {
      let url=MojConfig.adresa_servera+`/student/maticna-knjiga/${studentid}`;
      return this.httpClient.get<StudentMaticnaKnjigaGetResponse>(url);
    }
}

export interface StudentMaticnaKnjigaGetResponse {
  id: number,
  ime: string,
  prezime: string,
  upisaneGodine:StudentMaticnaKnjigaGetResponseUpisaneGodine[]
}

export interface StudentMaticnaKnjigaGetResponseUpisaneGodine{
  id: number,
  akademskaGodina: string,
  godinaStudija: string,
  obnova: boolean,
  zimskiSemestarUpis: Date,
  zimskiSemestarOvjera: Date | null,
  korisnikEvidentirao: StudentMaticnaKnjigaGetResponseUpisaneGodineKorisnik,
}

export interface StudentMaticnaKnjigaGetResponseUpisaneGodineKorisnik {
  id: number,
  ime: string,
  prezime: string,
}

File: back-end/FIT_Api_Example/Endpoints/MaticnaKnjigaEndpoints/Get/StudentMaticnaKnjigaGetEndpoint.cs
using FIT_Api_Example.Data;
using FIT_Api_Example.Helper;
using FIT_Api_Example.Helper.Auth;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace FIT_Api_Example.Endpoints.MaticnaKnjigaEndpoints.Get;

[Route("student/maticna-knjiga")]
[MyAuthorization]
public class StudentMaticnaKnjigaGetEndpoint : MyBaseEndpoint<int, StudentMaticnaKnjigaGetResponse>
{
    private readonly ApplicationDbContext _applicationDbContext;
    private readonly MyAuthService _authService;
    public StudentMaticnaKnjigaGetEndpoint(ApplicationDbContext applicationDbContext, MyAuthService authService)
    {
        _applicationDbContext = applicationDbContext;
        _authService = authService;
    }

    [HttpGet("{id}")]
    public override async Task<StudentMaticnaKnjigaGetResponse> Obradi(int id, CancellationToken cancellationToken)
    {
        var student = await _applicationDbContext.Student.FindAsync(id);

        if (student is null)
            throw new Exception("not found student for id = " + id);

        var result = new StudentMaticnaKnjigaGetResponse
        {

            Id = student.ID,
            Ime = student.Ime,
            Prezime = student.Prezime,
            UpisaneGodine = await _applicationDbContext
                .AkGodines
                .Where(x => x.StudentId == student.ID)
                .Select(x => new StudentMaticnaKnjigaGetResponseUpisaneGodine
                {
                    Id = x.Id,
                    AkademskaGodina = x.AkademskaGodina.Opis,
                    GodinaStudija = x.Godinastudina.ToString(),
                    KorisnikEvidentirao = new StudentMaticnaKnjigaGetResponseUpisaneGodineKorisnik
                    {
                        Id = x.EvidentiraoKorisnikId,
                        Ime = x.EvidentiraoKorisnik.KorisnickoIme,
                        Prezime = x.EvidentiraoKorisnik.KorisnickoIme
                    },
                    Obnova = x.JelObnova,
                    ZimskiSemestarUpis = x.DatumUpisZimski,
                    ZimskiSemestarOvjera = x.DatumOvjeraZimski
                }).ToListAsync(cancellationToken)
        };



        return result;
    }
}